features.bzl

```diff
@@ -33,6 +33,12 @@ _cc = struct(
     protobuf_on_allowlist = ge("8.0.0"),
 )
 
+_docs = struct(
+    # The stardoc output changed in https://github.com/bazelbuild/bazel/commit/bd1c3af2ea14e81268e940d2b8ba5ad00c3f08d7
+    # This may be required for "diff tests" that assert on the generated API docs.
```

Comment:
I don't want to make it too difficult to get a new feature in, but I wonder to what extent Stardoc guarantees output stability over time and how that aligns with major/minor releases of Bazel.

Would it make sense to only run diff tests on a single pinned Bazel version so that this kind of feature detection isn't needed? Happy to merge this if you still consider it useful.

Reply:
I agree. How do you think rules authors should change their code? Using https://github.com/bazel-contrib/rules-template/tree/main/docs as the example. Add `tags=[]` all over the BUILD file? Use an exclusion pattern like `-//docs:all` in their CI configuration for all but one Bazel version matrix?

Reply:
I tried a couple things and they're more complicated than this. The matrix expands to include nested modules, these don't have a `//docs` package so the exclusion pattern now needs to be conditional on that matrix entry, but it doesn't have a constant name (bazel version).

Reply:
The template probably isn't laid out in a way that makes this an easy change. Generally speaking, I nowadays try to keep all tests of external ruleset API in a separate Bazel module that is subject to a full Bazel version matrix. Internal tests are all Starlark analysis tests and tooling (including doc generation) and only run against the latest Bazel version. 

But yeah, it's just a boolean flag and it makes our lives easier, so let's merge it (will cut a release right after).